### PR TITLE
[IMP] web: enhance property field display in kanban view

### PR DESCRIPTION
**Pull request by vivp-odoo**
This PR improves the appearance of property fields in the Kanban view of
item Kanban. A label will be shown for the integer, float, date and datetime
property fields. Additionally, The boolean field now has a badge appearance
with a label.

**Task-3291930**

### CODE REVIEW

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
     <t t-name="web.KanbanPropertiesField" owl="1">
-        <div t-ref="properties" class="w-100">
+        <div t-ref="properties" class="w-100 mt-2">
             <div class="o_kanban_property_field d-inline-flex flex-column justify-content-center align-items-start w-100 my-1"
                 t-foreach="propertiesList"
                 t-as="propertyConfiguration"
                 t-key="propertyConfiguration.name"
-                t-if="(propertyConfiguration.value || ['integer', 'float'].includes(propertyConfiguration.type)) and propertyConfiguration.view_in_kanban">
-                <div class="mw-100 text-truncate">
+                t-if="propertyConfiguration.value &amp;&amp; propertyConfiguration.view_in_kanban">
```
Comment (schoffeniels) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:9]: Unless you have been asked to change it, I would keep it as it was before. Integers and floats may have `0` as value, which evaluate to `false`, but is a value and should be shown in my opinion.

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+                <div class="mw-100 text-truncate d-flex gap-2"
+                    t-att-class="(['boolean'].includes(propertyConfiguration.type)) ? 'border border-dark rounded-3 ps-2' : ''">
```
Comment (schoffeniels) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:11]: I would do a simple equality test here as the class is added only for booleans

**Review by schoffeniels — COMMENTED** [review]
Hello !

Looks good with a nice test, thanks for the work :)
Added a few small polishing comments

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml (hunk 1 above)
Comment (std-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:9]: keep using `and` and not `&amp;&amp;`

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml (hunk 2 above)
Comment (std-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:11]: you can use dict directly
```suggestion
                    t-att-class="{propertyConfiguration.type === 'boolean': 'border border-dark rounded-3 ps-2'}">
```

(or inverse key / value, I never sure about the synthax)

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 3]
```diff
@@ -1411,12 +1411,90 @@ QUnit.module("Fields", (hooks) => {
...
 
+    /**
+     * To check label for int, float, boolean, date and datetime fields.
+     *  Also check if border class is applied to boolean field or not.
+     */
+    QUnit.test("properties: kanban view with label and border", async function (assert) {
+        serverData.models.partner.records.push({
+            id: 12,
+            display_name: "fifth partner",
+            properties: [
```
Comment (std-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:1428]: can you apply the prettier rules in /web/tooling ?

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 4]
```diff
@@ -1411,12 +1411,90 @@ QUnit.module("Fields", (hooks) => {
...
+        });
+
+        // check for label in integer, float, date and datetime field
+        assert.equal(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(1) label").innerText,"My Integer:");
+        assert.equal(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(2) label").innerText,"My Float:");
+        assert.equal(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(3) label").innerText,"My Date:");
+        assert.equal(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(4) label").innerText,"My Datetime:");
+
+        //check that label and border class is present for checkbox field
+        assert.containsN(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(5)"), ".border",1);
```
Comment (std-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:1493]: containsOnce ?

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 5]
```diff
@@ -1411,12 +1411,90 @@ QUnit.module("Fields", (hooks) => {
...
+                            <field name="display_name"/> <hr/>
+                            <field name="properties" widget="properties"/>
+                        </div>
+                    </t>
+                </templates>
+            </kanban>`,
+        });
+
+        // check for label in integer, float, date and datetime field
+        assert.equal(target.querySelector(".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(1) label").innerText,"My Integer:");
```
Comment (std-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:1487]: I think `strictEqual` is better

**Review by std-odoo — COMMENTED** [review]
@vivp-odoo Hi :) 

Some comments while passing

Cheers

**Comment by vivp-odoo** [comment]
Hello @schoffeniels @std-odoo 
I have made the changes as suggested.
Please have a look at it.

Thanks!!

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
+        <div t-ref="properties" class="w-100 mt-2 fw-normal">
             <div class="o_kanban_property_field d-inline-flex flex-column justify-content-center align-items-start w-100 my-1"
                 t-foreach="propertiesList"
                 t-as="propertyConfiguration"
                 t-key="propertyConfiguration.name"
                 t-if="(propertyConfiguration.value || ['integer', 'float'].includes(propertyConfiguration.type)) and propertyConfiguration.view_in_kanban">
-                <div class="mw-100 text-truncate">
+                <div class="mw-100 text-truncate d-flex gap-2"
+                    t-att-class="{'border border-dark rounded-3 ps-2' : propertyConfiguration.type === 'boolean'}">
+                    <label t-if="['integer', 'float', 'date', 'datetime', 'boolean'].includes(propertyConfiguration.type)"><t t-out="propertyConfiguration.string"/>:</label>
```
Comment (std-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:12]: why `:` ? (generally we don't do that for label)

if you remove it the t-out can be on the <label> directly

**Review by std-odoo — COMMENTED** [review]
@vivp-odoo Hi :) 

I don't think you applied the prettier rules in /web/tooling from my previous review :/ 

screenshot_spect

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml (hunk 6 above)
Comment (vivp-odoo) [reply addons/web/static/src/views/fields/properties/kanban_properties_field.xml:12]: I have used **:** as per the suggestion of the PO team to clearly distinguish between label and value

**Comment by vivp-odoo** [comment]
Hello @std-odoo @schoffeniels 
made the changes.
please have a look at it.

Thanks!!

**Comment by dbeguin** [comment]
@robodoo r+

File: addons/web/static/src/views/fields/properties/kanban_properties_field.scss [hunk 7]
```diff
@@ -1,3 +1,11 @@
 .o_kanban_property_field {
     max-width: 300px;
 }
+
+.form-check{
+    margin-bottom: 0;
+}
+
+.o_kanban_property_field{
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.scss:9]: Why are we duplicating this rule?

File: addons/web/static/src/views/fields/properties/kanban_properties_field.scss [hunk 8]
```diff
@@ -1,3 +1,11 @@
 .o_kanban_property_field {
     max-width: 300px;
 }
+
+.form-check{
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.scss:5]: Doesn't that apply to every single form-check class in the entire world? Would need to be included inside a more specific rule ideally.

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 9]
```diff
@@ -1,13 +1,17 @@
...
     <t t-name="web.KanbanPropertiesField" owl="1">
-        <div t-ref="properties" class="w-100">
+        <div t-ref="properties" class="w-100 mt-2 fw-normal text-muted">
             <div class="o_kanban_property_field d-inline-flex flex-column justify-content-center align-items-start w-100 my-1"
                 t-foreach="propertiesList"
                 t-as="propertyConfiguration"
                 t-key="propertyConfiguration.name"
-                t-if="(propertyConfiguration.value || ['integer', 'float'].includes(propertyConfiguration.type)) and propertyConfiguration.view_in_kanban">
-                <div class="mw-100 text-truncate">
+                t-if="propertyConfiguration.value and propertyConfiguration.view_in_kanban">
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:9]: I see this comment marked as resolved:

> Unless you have been asked to change it, I would keep it as it was before. Integers and floats may have 0 as value, which evaluate to false, but is a value and should be shown in my opinion.

But no answer to it, isn't that a real problem?

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 10]
```diff
@@ -1,13 +1,17 @@
...
+                <div class="mw-100 text-truncate d-flex gap-2"
+                    t-att-class="{'border border-dark rounded-3 ps-2' : propertyConfiguration.type === 'boolean'}">
+                    <label t-if="['integer', 'float', 'date', 'datetime', 'boolean'].includes(propertyConfiguration.type)"
+                    t-att-class="{'fw-bold' : propertyConfiguration.type != 'boolean'}">
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:13]: Missing one indentation level.

Also should be `!==` as it's a JS comparison.

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 11]
```diff
@@ -1,13 +1,17 @@
...
+        <div t-ref="properties" class="w-100 mt-2 fw-normal text-muted">
             <div class="o_kanban_property_field d-inline-flex flex-column justify-content-center align-items-start w-100 my-1"
                 t-foreach="propertiesList"
                 t-as="propertyConfiguration"
                 t-key="propertyConfiguration.name"
-                t-if="(propertyConfiguration.value || ['integer', 'float'].includes(propertyConfiguration.type)) and propertyConfiguration.view_in_kanban">
-                <div class="mw-100 text-truncate">
+                t-if="propertyConfiguration.value and propertyConfiguration.view_in_kanban">
+                <div class="mw-100 text-truncate d-flex gap-2"
+                    t-att-class="{'border border-dark rounded-3 ps-2' : propertyConfiguration.type === 'boolean'}">
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:11]: Strange way to write this, is this a ternary operator with only one side?

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 12]
```diff
@@ -1,13 +1,17 @@
...
+                    <label t-if="['integer', 'float', 'date', 'datetime', 'boolean'].includes(propertyConfiguration.type)"
+                    t-att-class="{'fw-bold' : propertyConfiguration.type != 'boolean'}">
+                    <t t-out="propertyConfiguration.string"/></label>
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:14]: The t-out can be inside the label tag to save some code.

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 13]
```diff
@@ -1411,12 +1411,120 @@ QUnit.module("Fields", (hooks) => {
...
+                    "Company 2 other partner My Integer1",
                 ]
             );
         }
     );
 
+    /**
+     * To check label for int, float, boolean, date and datetime fields.
+     *  Also check if border class is applied to boolean field or not.
+     */
```
Comment (awa-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:1423]: Comment should ideally be glued to the next part, remove the blank space.

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 14]
```diff
@@ -1411,12 +1411,120 @@ QUnit.module("Fields", (hooks) => {
...
+            "My Datetime"
+        );
+
+        //check that label and border class is present for checkbox field
+        assert.containsOnce(
+            target.querySelector(
+                ".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(5)"
+            ),
+            ".border"
+        );
```
Comment (awa-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:1519]: Can't you simplify this to:

`assert.containsOnce(target, ".o_kanban_record:nth-child(5) .o_kanban_property_field:nth-child(5) .border");`

?

**Review by awa-odoo — COMMENTED** [review]
Hello @vivp-odoo 

Some additional polishing comments.

Thanks & cheers!

**Comment by awa-odoo** [comment]
@robodoo r-

(would like to polish a bit more first)

File: addons/web/static/src/views/fields/properties/kanban_properties_field.scss [hunk 15]
```diff
@@ -1,3 +1,9 @@
 .o_kanban_property_field {
     max-width: 300px;
+    font-size: 0.875rem;
+
+    .form-check{
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.scss:5]: (Linting: Also missing a space after the rule name and before the opening bracket)

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml (hunk 9 above)
Comment (vivp-odoo) [reply addons/web/static/src/views/fields/properties/kanban_properties_field.xml:9]: @awa-odoo 
This issue has been resolved over task pad. The PO agreed to change it. we won't show integers and float for 0 value.

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml (hunk 11 above)
Comment (vivp-odoo) [reply addons/web/static/src/views/fields/properties/kanban_properties_field.xml:11]: @awa-odoo this was suggested by @std-odoo.

File: addons/web/static/src/views/fields/properties/kanban_properties_field.xml [hunk 16]
```diff
@@ -1,13 +1,16 @@
...
+        <div t-ref="properties" class="w-100 mt-2 fw-normal text-muted">
             <div class="o_kanban_property_field d-inline-flex flex-column justify-content-center align-items-start w-100 my-1"
                 t-foreach="propertiesList"
                 t-as="propertyConfiguration"
                 t-key="propertyConfiguration.name"
-                t-if="(propertyConfiguration.value || ['integer', 'float'].includes(propertyConfiguration.type)) and propertyConfiguration.view_in_kanban">
-                <div class="mw-100 text-truncate">
+                t-if="propertyConfiguration.value and propertyConfiguration.view_in_kanban">
+                <div class="mw-100 text-truncate d-flex gap-2"
+                    t-att-class="{'border border-dark rounded-3 ps-2' : propertyConfiguration.type === 'boolean'}">
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/properties/kanban_properties_field.xml:11]: While reviewing this with the PO, he told me we can remove "border-dark", as the lighter colored border looks nicer in the UI.

**Review by awa-odoo — COMMENTED** [review]
Hello @vivp-odoo 

A couple more things and then we should be good to go :rocket: 

Thanks for the followup and cheers!

**Comment by awa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**